examples/multi_gpu/data_parallel.py

```diff
@@ -53,7 +53,12 @@ def forward(self, data):
 model = Net()
 print(f"Let's use {torch.cuda.device_count()} GPUs!")
 model = DataParallel(model)
-device = torch.device('cuda:0' if torch.cuda.is_available() else 'cpu')
+if torch.cuda.is_available():
+    device = torch.device('cuda:0')
+elif hasattr(torch.backends, 'mps') and torch.backends.mps.is_available():
+    device = torch.device('mps')
+else:
+    device = torch.device('cpu')
```

Comment:
> I have tested every example one by one on MPS background on my apple silicon mac. Everything works on GPU Acceleration

@NripeshN Are you sure everything works? Data Parallel doesn't work on Apple Silicon afaik.

Reply:
Hi, it worked fine on my laptop without throwing any errors but this program did not run on my friend's laptop. I am not entirely sure why.

Reply:
@akihironitta @rusty1s I am sorry for my comment earlier, I had written a script that would run all Python scripts in a directory and print out the files that throw out errors. Unfortunately that program issues and showed everything passed.